pkg/pki/certmanagerpki/certmanager_pki.go

```diff
@@ -20,6 +22,62 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
 )
 
+func (c *certManager) GetCertificate(ctx context.Context, logger zap.Logger) (certv1.Certificate, error) {
+	cert := &certv1.Certificate{}
+
+	objName := types.NamespacedName{
+		Name:      fmt.Sprintf(pkicommon.NodeCACertTemplate, c.cluster.Name),
```

Comment:
It looks like this `GetCertificate()` method only returns the cluster CA certificate, which nifikop only uses when it is self-signing certificates. For example, this certificate is only used when you don't provide a `ClusterIssuer`/`Issuer` to sign certificates. 

Was it intentional to not watch each node's certificate expiration date? If configured, node certs are generated by the pki manager creating `NifiUser` CRDs:

https://github.com/konpyutaika/nifikop/blob/master/pkg/util/pki/common.go#L162-L194

Then as a part of the `NifiUser` reconciliation, PKI is generated:

https://github.com/konpyutaika/nifikop/blob/master/pkg/pki/certmanagerpki/certmanager_user.go#L28-L69